docs: misc docs reorganize and cleanups (#2768)
bentoml/BentoML: docs/source/frameworks/index.rst

Framework Guides
================

Here is the list of supported ML libraries and formats in BentoML. You can also find example
projects in the `bentoml/gallery <https://github.com/bentoml/gallery>`_ repository.

.. grid:: 1 2 2 2
    :gutter: 3
    :margin: 0
    :padding: 3 4 0 0

>     .. grid-item-card:: :doc:`/frameworks/catboost`

Author: reordered alphabetically

        :link: /frameworks/catboost
        :link-type: doc

    .. grid-item-card:: :doc:`/frameworks/lightgbm`
        :link: /frameworks/lightgbm
        :link-type: doc

Custom Models
-------------

.. grid:: 1 2 2 2
    :gutter: 3
    :margin: 0
    :padding: 3 4 0 0

    .. grid-item-card:: :doc:`/frameworks/picklable`
        :link: /frameworks/picklable
        :link-type: doc

    .. grid-item-card:: :ref:`concepts/runner:Custom Runner`
        :link: /concepts/runner
        :link-type: doc

.. TODO:
    .. grid-item-card:: :doc:`/guides/custom_ml_framework`
        :link: /guides/custom_ml_framework
        :link-type: doc

Roadmap
-------

> The following frameworks are supported in pre-1.0 BentoML versions and they are
> still being migrated to the new API in 1.0. Before they are officially supported,
> users may use :ref:`Custom Models <frameworks/index:Custom Models>` as an workaround.

Reviewer: ```suggestion
The following frameworks are supported in pre-1.0 BentoML versions and are being migrated to the new 1.0 API. In the meantime, users may use :ref:`Custom Models <frameworks/index:Custom Models>` as a workaround.
```

        :link: /frameworks/xgboost
        :link-type: doc

> Custom Models

Author: moved out pickable and custom runner for "custom models"

will add an additional guide for custom ml framework in bentoml later on

.. toctree::
    :hidden:

>     catboost

Author: reordered alphbetically
